src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1086,6 +1088,16 @@ public SubmoduleUpdateCommand timeout(Integer timeout) {
                 return this;
             }
 
+            public SubmoduleUpdateCommand shallow(boolean shallow) {
+                this.shallow = shallow;
+                return this;
+            }
+
+            public SubmoduleUpdateCommand depth(Integer depth) {
```

Comment:
If possible I would go for `int` instead of `Integer` for the method parameter and instance variable. (I also see `boolean` and no `Boolean`).

Reply:
Ok. I have seen now that this is done like this to be consisent with the methods for clone/fetch.